internal/venice-common/src/main/java/com/linkedin/venice/acl/handler/StoreAclHandler.java

```diff
@@ -48,15 +49,22 @@ protected String extractStoreName(String resourceName) {
     return resourceName;
   }
 
-  protected X509Certificate extractClientCert(ChannelHandlerContext ctx) throws SSLPeerUnverifiedException {
+  protected X509Certificate extractClientCert(ChannelHandlerContext ctx) {
     SslHandler sslHandler = ctx.pipeline().get(SslHandler.class);
     if (sslHandler == null) {
       /**
        * In HTTP/2, the SSLHandler is k  parent channel pipeline and the child channels won't have the SSL Handler.
        */
       sslHandler = ctx.channel().parent().pipeline().get(SslHandler.class);
     }
-    return SslUtils.getX509Certificate(sslHandler.engine().getSession().getPeerCertificates()[0]);
+    final Certificate[] peerCertificates;
+    try {
+      peerCertificates = sslHandler.engine().getSession().getPeerCertificates();
+    } catch (SSLPeerUnverifiedException exception) {
+      return null;
```

Comment:
This I'm a bit less sure about.  The intention of this server handler is such that it it's only actually part of the request pipeline if ACL's are enabled.  If ACL's are enabled, then we want this thing to be able to extract identity's of actors and scream at us if it can't do that.

Now, what's bothersome today is that it's taken for granted that the only way to determine the identity of a requester is via client certificate.  What'd be ideal is to make that identity extraction pluggable (so instead of it being strictly peer cert, maybe it's a crypto header or something idk).

So, my thinking is that we can maybe leave this alone for now with the intention of a bigger refactor based on @eolivelli's vip and you can get unblocked by just not enabling ACLs?